src/BDD/BitSet.jl

```diff
@@ -0,0 +1,77 @@
+"""
+    mutable struct BitSet <: AbstractSet{Int}
+
+Same as `Base.BitSet` but with `CUDD`.
+"""
+mutable struct BitSet <: AbstractSet{Int}
+    manager::Ptr{CUDD.DdManager}
+    variables::Vector{Ptr{CUDD.DdNode}}
+    root::Ptr{CUDD.DdNode}
+    function BitSet()
+        manager = CUDD.initialize_cudd()
+        return new(manager, Ptr{CUDD.DdNode}[], CUDD.Cudd_ReadLogicZero(manager))
+    end
+end
+
+Base.show(io::IO, set::BitSet) = Base.summary(io, set)
+function Base.summary(io::IO, set::BitSet)
+    n = length(set.variables)
+    Base.showarg(io, set, true)
+    print(io, " with ", n, " bit")
+    if !isone(n)
+        print(io, "s")
+    end
+end
+
+Base.eltype(::Type{BitSet}) = Int
+Base.empty(::BitSet, ::Type{Int}=Int) = BitSet()
+Base.emptymutable(::BitSet, ::Type{Int}=Int) = BitSet()
+Base.isempty(set::BitSet) = set.root === CUDD.Cudd_ReadLogicZero(set.manager)
+function Base.empty!(set::BitSet)
+    set.root = CUDD.Cudd_ReadLogicZero(set.manager)
+    return set
+end
+Base.IteratorSize(::BitSet) = Base.SizeUnknown()
+
+function Base.iterate(set::BitSet, state::Int=0)
+    phase, x = phase_rem(set.variables, state)
+    if iszero(x)
+        if _in(phase, set)
+            return (state, state + 1)
+        else
+            return iterate(set, state + 1)
+        end
+    else
+        # state would require more variables than what is in `set.variables`
+        # so it is necessarily not in `set`.
+        return nothing
+    end
+end
+
+function phase!(set::BitSet, x::Int)
+    phase, x = phase_rem(set.variables, x)
+    while x > 0
+        push!(phase, iszero(x & 1) ? zero(Cint) : one(Cint))
+        # As the `manager` is only used by this struct, `bddIthVar` should be
+        # the same as `bddAddVar`.
+        var = CUDD.Cudd_bddIthVar(set.manager, Cint(length(set.variables)))
```

Comment:
Why not `Cudd_bddNewVar(set.manager)`?

Reply:
So that it works even if the user plays with the internal manager :-P